flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -36,52 +35,49 @@
 
 import static org.apache.flink.runtime.state.ChannelPersistenceITCase.getStreamFactoryFactory;
 import static org.apache.flink.util.CloseableIterator.ofElements;
-import static org.apache.flink.util.ExceptionUtils.findThrowable;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 /** {@link ChannelStateWriterImpl} lifecycle tests. */
-public class ChannelStateWriterImplTest {
+class ChannelStateWriterImplTest {
     private static final long CHECKPOINT_ID = 42L;
     private static final String TASK_NAME = "test";
 
-    @Test(expected = IllegalArgumentException.class)
-    public void testAddEventBuffer() throws Exception {
+    @Test
+    void testAddEventBuffer() throws Exception {
 
         NetworkBuffer dataBuf = getBuffer();
         NetworkBuffer eventBuf = getBuffer();
         eventBuf.setDataType(Buffer.DataType.EVENT_BUFFER);
-        try {
-            runWithSyncWorker(
-                    writer -> {
-                        callStart(writer);
-                        writer.addInputData(
-                                CHECKPOINT_ID,
-                                new InputChannelInfo(1, 1),
-                                1,
-                                ofElements(Buffer::recycleBuffer, eventBuf, dataBuf));
-                    });
-        } finally {
-            assertTrue(dataBuf.isRecycled());
-        }
+
+        runWithSyncWorker(
+                (writer, worker) -> {
+                    callStart(writer);
+                    callAddInputData(writer, dataBuf);
```

Comment:
Yes , I'm sure. 

Channel state just allows the data buffer, and doesn't allow the event buffer. This unit-test checks whether it throw `IllegalArgumentException` when adding an event buffer. 

So I call `processAllRequests` twice, the first one, it's a `start request` and `add data buffer request`, the second one, it's a `add event buffer request`. 

The first one is normal, and the second one should throw exception. It's more specific.

Reply:
I guess, you're right. But the test is slightly changed by you adding `dataBuf` before adding `eventBuf` because we're checking whether `dataBuf` is recycled afterwards. The semantics of this check are changed due to your change: Previously, the test checked that a buffer was recycled which should have been processed but wasn't processed because of the Exception the `eventBuf` caused. Now, we check that the `dataBuf` is recycled after being processed properly. I'm curious about your opinion here.

Reply:
> he test checked that a buffer was recycled which should have been processed but wasn't processed because of the Exception the eventBuf caused. 

Good catch, I didn't realize it before. I just thought that test wanted to verify whether the data buffer is normal.

Now I understand it, thanks for your reminder~

Reply:
Thanks for the quick response, @1996fanrui. Looking at this specific test, we will see now that we run into the same issue which I explained in [my other comment](https://github.com/apache/flink/pull/21368#discussion_r1039276265). The expected `IllegalArgumentException` is caught and handled by `assertThatThrownBy`. As a consequence, the call chain continues within `runWithSyncWorker` making `processAllRequests` being called once more which might affect `dataBuf` in this specific case. I'm gonna propose a way to deal with this in the other thread to keep that discussion there.